src/v0/destinations/klaviyo/transform.js

```diff
@@ -42,61 +52,78 @@ const { ConfigurationError, InstrumentationError } = require('../../util/errorTy
   // Extract other K-V property from traits about user custom properties
-  propertyPayload = extractCustomFields(
+  let customPropertyPayload = {};
+  customPropertyPayload = extractCustomFields(
     message,
-    propertyPayload,
+    customPropertyPayload,
     ['traits', 'context.traits'],
     WhiteListedTraits,
   );
-  if (!personId) {
-    propertyPayload = removeUndefinedAndNullValues(propertyPayload);
-    if (destination.Config?.enforceEmailAsPrimary) {
-      delete propertyPayload.$id;
-      propertyPayload._id = getFieldValueFromMessage(message, 'userId');
-    }
-    const payload = {
-      token: destination.Config.publicApiKey,
-      properties: propertyPayload,
-    };
-    response.endpoint = `${BASE_ENDPOINT}${category.apiUrl}`;
-    response.method = defaultPostRequestConfig.requestMethod;
-    response.headers = {
-      'Content-Type': 'application/json',
-      Accept: 'text/html',
-    };
-    response.body.JSON = removeUndefinedAndNullValues(payload);
-  } else {
-    response.endpoint = `${BASE_ENDPOINT}/api/v1/person/${personId}`;
-    response.method = defaultPutRequestConfig.requestMethod;
-    response.headers = {
-      Accept: 'application/json',
-    };
-    response.params = removeUndefinedAndNullValues(propertyPayload);
-    response.params.api_key = destination.Config.privateApiKey;
+
+  propertyPayload = removeUndefinedAndNullValues(propertyPayload);
+  if (enforceEmailAsPrimary) {
+    delete propertyPayload.external_id;
+    propertyPayload.properties._id = getFieldValueFromMessage(message, 'userId');
+  }
+  const data = {
+    type: 'profile',
+    attributes: propertyPayload,
+    properties: removeUndefinedAndNullValues(customPropertyPayload),
+  };
+  const payload = {
+    data: removeUndefinedAndNullValues(data),
+  };
+  const endpoint = `${BASE_ENDPOINT}${category.apiUrl}`;
+  const requestOptions = {
+    headers: {
+      Authorization: `Klaviyo-API-Key ${privateApiKey}`,
+      accept: 'application/json',
+      'content-type': 'application/json',
+      revision: '2023-02-22',
+    },
+  };
+  const resp = await httpPOST(endpoint, payload, requestOptions);
+  if (resp.response?.status === 201) {
+    profileId = resp.response.data.id;
+  } else if (resp.response?.response?.status === 409) {
+    const { response } = resp.response;
+    profileId = response.data.errors[0].meta.duplicate_profile_id;
+  }
+
+  // Update Profile
+  const identifyResponse = defaultRequestConfig();
+  payload.data.id = profileId;
+  identifyResponse.endpoint = `${BASE_ENDPOINT}${category.apiUrl}/${profileId}`;
+  identifyResponse.method = defaultPatchRequestConfig.requestMethod;
+  identifyResponse.headers = {
+    Authorization: `Klaviyo-API-Key ${privateApiKey}`,
+    'Content-Type': 'application/json',
+    Accept: 'application/json',
+    revision: '2023-02-22',
+  };
+  identifyResponse.body.JSON = removeUndefinedAndNullValues(payload);
+  const responseArray = [identifyResponse];
```

Comment:
Do we need to make this call in the situation when the user is getting created for the first time?

Reply:
Yes